Add routing

jzpnhlim opened the pull request:
recommit the json Valhalla code, removed hardcoded IP

added the json nominatim code to convert a room to its longitude latitude

jzpnhlim committed:
add routing

jzpnhlim committed:
add function that find long/lat by room

mostafaayesh committed:
replace http with dio

mostafaayesh committed:
Fix typo in nominatim.dart

MostafaAyesh commented:
Changes I made:
- Replace HTTP with Dio to simplify structuring requests
- Created GlobalCoordinates and LocalCoordinates classes
- Add a test case for Nominatim

Some changes to be done:
- Improve the Valhalla module (A class has to be structured in way providing access to the different components of the response. The function to fetch the route should return an object of that class providing access to the shape, instructions and other relevant information)
- Improve the Nominatim module (Provide more functions, eg. function to find a list of locations instead of just the coordinates for one location)
- Add test cases for Valhalla and Nominatim

jzpnhlim committed:
improves valhalla module

imporved valhalla module

fetch route function now returns the full json response.

add getShape() funciton to get polyline

add getInstruction() function to get the instruction set

add test cases for valhalla module

jzpnhlim committed:
correct class type mismatch

jzpnhlim committed:
get all access to nominatim output

added seperate getGlobalCoordinates(NominatimOutput) function

tested

jzpnhlim committed:
fixed type conflict and add funciton convert val instruction to instruciton type

tested

jzpnhlim committed:
fixed minor issue

fixed issue where the instruction has a movement with 0 meters at the very end which is unnessary to display

SteveLoveWork committed:
Tried to update the instruction base on the rounting

- Tried to get the instruction from the routing command and get it to update on the instruction screen
